cli/src/check.rs

```diff
@@ -0,0 +1,217 @@
+use k8s_openapi::{
+    api::{apps::v1::Deployment, core::v1::Namespace},
+    apiextensions_apiserver::pkg::apis::apiextensions::v1::CustomResourceDefinition,
+};
+use kube::{api::ListParams, Api, Client, ResourceExt};
+use serde::Serialize;
+
+const CHECK: &str = "√";
+const EX: &str = "×";
+
+#[derive(Serialize)]
+struct CheckOutput {
+    success: bool,
+    categories: Vec<Category>,
+}
+
+#[derive(Serialize)]
+struct Category {
+    category_name: &'static str,
+    checks: Vec<CheckResult>,
+}
+
+#[derive(Serialize)]
+#[serde(rename_all = "lowercase")]
+enum CheckStatus {
+    Success,
+    Error,
+}
+
+#[derive(Serialize, Default)]
+pub struct CheckResult {
+    description: &'static str,
+    result: CheckStatus,
+    #[serde(skip_serializing_if = "Option::is_none")]
+    hint: Option<&'static str>,
+    #[serde(skip_serializing_if = "Option::is_none")]
+    error: Option<String>,
+}
+
+impl Default for CheckStatus {
+    fn default() -> CheckStatus {
+        CheckStatus::Success
+    }
+}
+
+pub async fn traffic_split_check(client: Client) -> CheckResult {
+    let api = Api::<CustomResourceDefinition>::all(client.clone());
+    let description = "TrafficSplit CRD exists";
```

Comment:
We could `const` all of the descriptions:
```suggestion
    const DESCRIPTION: &str = "TrafficSplit CRD exists";
```

(It would obviously require updating the `CheckResult`s below

Reply:
FWIW, I _believe_ the `let`-binding here probably gets const-folded regardless, but this might be nicer to be more explicit about...

Reply:
the let binding is immutable, so does marking these as const communicate anything useful?

Reply:
it's fine either way but the intent is clearer: this is a constant vs 'this is a variable.' let bindings can be rebound, for instance:

```
let foo = "foo";
let foo = "bar";
``` 

They can also be consumed.

In this case there's no difference, but in general there are differences.